### PR TITLE
internal: log when connection is closed due to keepalive

### PR DESCRIPTION
**Pull request by menghanl**

### CODE REVIEW

File: internal/transport/http2_server.go [hunk 1]
```diff
@@ -965,6 +965,7 @@ func (t *http2Server) keepalive() {
 			select {
 			case <-maxAge.C:
 				// Close the connection after grace period.
+				infof("transport: closing server transport due to maxAge.")
```
Comment (dfawley) [inline internal/transport/http2_server.go]: "maximum connection age (configuration?)"?

File: internal/transport/http2_server.go [hunk 2]
```diff
@@ -978,6 +979,7 @@ func (t *http2Server) keepalive() {
 				continue
 			}
 			if pingSent {
+				infof("transport: closing server transport due to keepalive.")
```
Comment (dfawley) [inline internal/transport/http2_server.go]: as above.

File: internal/transport/http2_client.go [hunk 3]
```diff
@@ -1332,6 +1332,7 @@ func (t *http2Client) keepalive() {
 					timer.Reset(t.kp.Time)
 					continue
 				}
+				infof("transport: closing client transport due to keepalive.")
```
Comment (dfawley) [inline internal/transport/http2_client.go]: "due to idleness / lack of connection activity"?

File: internal/transport/http2_server.go (hunk 1 above)
Comment (menghanl) [reply internal/transport/http2_server.go]: Done

File: internal/transport/http2_server.go (hunk 2 above)
Comment (menghanl) [reply internal/transport/http2_server.go]: Done

File: internal/transport/http2_client.go (hunk 3 above)
Comment (menghanl) [reply internal/transport/http2_client.go]: Done

**Pull request merged**